src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,90 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_CAPACITY];
+        this.threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
+
+    private static class Node<K, V> {
+        final K key;
+        V value;
+        Node<K, V> next;
+
+        Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
+        int index = getIndex(key);
+        Node<K, V> currentNode = table[index];
+
+        while (currentNode != null) {
+            if (currentNode.key == key || (key != null && key.equals(currentNode.key))) {
+                currentNode.value = value;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
+
+        Node<K, V> newNode = new Node<>(key, value, table[index]);
+        table[index] = newNode;
+        size++;
 
+        if (size >= threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> currentNode = table[index];
+
+        while (currentNode != null) {
+            if (currentNode.key == key || (key != null && key.equals(currentNode.key))) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode() % table.length);
```

Comment:
The `getIndex` method should handle negative hash codes correctly. Consider using `Math.abs(key.hashCode()) % table.length` to ensure the index is always non-negative. However, be cautious with `Math.abs` as it can return negative values for `Integer.MIN_VALUE`. A safer approach might be `(key.hashCode() & 0x7fffffff) % table.length`.